src/config.ts

```diff
@@ -488,3 +499,24 @@ export function mergeStyle(
 export function stringOrNull(spec: unknown): string | null {
   return spec == null || spec === false ? null : String(spec);
 }
+
+function normalizeDuckDB(spec: unknown): DuckDBConfig {
+  const install = spec?.["install"] ?? ["json", "parquet"];
+  const load = spec?.["load"] ?? [];
+  const source = new Map(Object.entries(spec?.["source"] ?? {}));
+  return {
+    bundles: ["eh", "mvp"],
+    install,
+    load: load.filter((name: string) => install.includes(name)),
+    source: Object.fromEntries(
+      install.map((name: string) => {
+        let href = source.get(name) ?? "core";
+        if (href === "core") href = "https://extensions.duckdb.org";
+        else if (href === "community") href = "https://community-extensions.duckdb.org";
+        if (!href?.["startsWith"]?.("https://"))
+          throw new Error(`unknown source for duckdb extension ${name}: ${href}`);
```

Comment:
What if you want to load an extension that you’ve manually copied into your source root?

Reply:
Maybe possible if you said `import.meta.resolve("duckdb:local:foo")`? But this is probably so rare that we don’t need to support this from the get-go.